packages/core/strapi/lib/core/registries/cron.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const cron = require('node-schedule');
+const { isFunction, getOr } = require('lodash/fp');
+
+const cronRegistry = () => {
+  let jobs = [];
+
+  return {
+    add(tasks = {}) {
+      if (!strapi.config.get('server.cron.enabled')) {
+        return;
+      }
+
+      for (const taskExpression in tasks) {
+        const taskValue = tasks[taskExpression];
+
+        if (isFunction(taskValue)) {
+          const instanciatedTask = taskValue(strapi);
+          return cron.scheduleJob(taskExpression, instanciatedTask);
+        }
+
+        const options = getOr({}, 'options', taskValue);
+
+        const job = cron.scheduleJob(
+          {
+            rule: taskExpression,
+            ...options,
+          },
+          taskValue.task(strapi)
+        );
+        jobs.push(job);
+      }
+    },
+    cancelAll() {
```

Comment:
can we make that generic and call it destroy ? as an abstraction we just want to destroy the cron service whatever it does to clean itself up is not relevant outside.